src/sagemaker/workflow/model_step.py

```diff
@@ -0,0 +1,198 @@
+                ".create() or .register() method to generate the step_args under PipelineSession."
+            )
+        if not isinstance(step_args.model.sagemaker_session, PipelineSession):
+            raise TypeError(
+                "To correctly configure a ModelStep, "
+                "the sagemaker_session of the model must be a PipelineSession object."
+            )
+
+        self.name = name
+        self.step_args = step_args
+        self.depends_on = depends_on
+        self.retry_policies = retry_policies
+        self.display_name = display_name
+        self.description = description
+        self.steps: List[Step] = []
+        self._model = step_args.model
+        self._create_model_args = self.step_args.create_model_request
+        self._register_model_args = self.step_args.create_model_package_request
+        self._need_runtime_repack = self.step_args.need_runtime_repack
+
+        if isinstance(retry_policies, dict):
+            self._create_model_retry_policies = retry_policies.get(
+                _CREATE_MODEL_RETRY_POLICIES, None
+            )
+            self._register_model_retry_policies = retry_policies.get(
+                _REGISTER_MODEL_RETRY_POLICIES, None
+            )
+            self._repack_model_retry_policies = retry_policies.get(
+                _REPACK_MODEL_RETRY_POLICIES, None
+            )
+        else:
+            self._create_model_retry_policies = retry_policies
+            self._register_model_retry_policies = retry_policies
+            self._repack_model_retry_policies = retry_policies
+        self._repack()
+
+        if self._register_model_args:
+            self._append_register_model_step()
+        else:
+            self._append_create_model_step()
+
+    def _append_register_model_step(self):
+        """Create and append a `_RegisterModelStep`"""
+        register_model_step = _RegisterModelStep(
+            name="{}-{}".format(self.name, _REGISTER_MODEL_NAME_BASE),
+            step_args=self._register_model_args,
+            display_name=self.display_name,
+            retry_policies=self._register_model_retry_policies,
+            description=self.description,
+        )
+        if not self._need_runtime_repack:
+            register_model_step.add_depends_on(self.depends_on)
+        self.steps.append(register_model_step)
+
+    def _append_create_model_step(self):
+        """Create and append a `CreateModelStep`"""
+        create_model_step = CreateModelStep(
+            name="{}-{}".format(self.name, _CREATE_MODEL_NAME_BASE),
+            step_args=self._create_model_args,
+            retry_policies=self._create_model_retry_policies,
+            display_name=self.display_name,
+            description=self.description,
+        )
+        if not self._need_runtime_repack:
+            create_model_step.add_depends_on(self.depends_on)
+        self.steps.append(create_model_step)
+
+    def _repack(self):
```

Comment:
nit: not sure if its best to follow pattern for naming here? `_append_repack_model_step`

Reply:
Good call! 
Yea, the `_repack` may be confusing as people may expect the repack happens here in the SDK compile time.
Updated in the new commit.